rasa/nlu/featurizers/sparse_featurizer/count_vectors_featurizer.py

```diff
@@ -211,184 +210,178 @@ def __init__(
-        return text
+                tokens = [self.OOV_token if t in self.OOV_words else t for t in tokens]
 
-    @staticmethod
-    def _get_message_tokens_by_attribute(
-        message: "Message", attribute: Text
+        return tokens
+
+    def _get_processed_message_tokens_by_attribute(
+        self, message: "Message", attribute: Text = TEXT_ATTRIBUTE
     ) -> List[Text]:
-        """Get text tokens of an attribute of a message"""
+        """Get processed text of attribute of a message"""
+
+        if message.get(attribute) is None:
+            # return empty string since sklearn countvectorizer does not like None
+            # object while training and predicting
+            return [""]
+
+        tokens = self._get_message_tokens_by_attribute(message, attribute)
+        tokens = self._process_tokens(tokens, attribute)
+        tokens = self._replace_with_oov_token(tokens, attribute)
 
-        if attribute in SPACY_FEATURIZABLE_ATTRIBUTES and message.get(
-            MESSAGE_SPACY_FEATURES_NAMES[attribute]
-        ):  # if lemmatize is possible
-            tokens = [
-                t.lemma_ for t in message.get(MESSAGE_SPACY_FEATURES_NAMES[attribute])
-            ]
-        elif message.get(
-            MESSAGE_TOKENS_NAMES[attribute]
-        ):  # if directly tokens is provided
-            tokens = [t.text for t in message.get(MESSAGE_TOKENS_NAMES[attribute])]
-        else:
-            tokens = message.get(attribute).split()
         return tokens
 
     # noinspection PyPep8Naming
-    def _check_OOV_present(self, examples) -> None:
+    def _check_OOV_present(self, all_tokens: List[List[Text]]):
         """Check if an OOV word is present"""
-        if self.OOV_token and not self.OOV_words:
-            for t in examples:
-                if (
-                    t is None
-                    or self.OOV_token in t
-                    or (self.lowercase and self.OOV_token in t.lower())
-                ):
-                    return
-
-            warnings.warn(
-                f"OOV_token='{self.OOV_token}' was given, but it is not present "
-                "in the training data. All unseen words "
-                "will be ignored during prediction."
-            )
-
-    def _set_attribute_features(
-        self,
-        attribute: Text,
-        attribute_features: np.ndarray,
-        training_data: "TrainingData",
-    ):
-        """Set computed features of the attribute to corresponding message objects"""
-        for i, example in enumerate(training_data.intent_examples):
-            # create bag for each example
-            example.set(
-                MESSAGE_VECTOR_FEATURE_NAMES[attribute],
-                self._combine_with_existing_features(
-                    example,
-                    attribute_features[i],
-                    MESSAGE_VECTOR_FEATURE_NAMES[attribute],
-                ),
-            )
+        if self.OOV_token and not self.OOV_words and all_tokens:
```

Comment:
I'd us an early exit here (to decrease indentation and make this easier to understand):
```
        if not self.OOV_token or self.OOV_words or not all_tokens:
            return

```